src/eeprom24x.rs

```diff
@@ -228,9 +228,30 @@ macro_rules! impl_for_page_size {
             }
         }
 
+        impl<I2C, E, AS> PageWrite<E> for Eeprom24x<I2C, page_size::$PS, AS>
+        where
+            I2C: Write<Error = E>,
+            AS: MultiSizeAddr,
+        {
+            fn page_write(&mut self, address: u32, data: &[u8]) -> Result<(), Error<E>> {
+                self.write_page(address, data)
+            }
+
+            fn page_size(&self) -> usize {
+                $page_size
+            }
+        }
+
     };
 }
 
+/// Helper trait which gives the Storage implementation access to the `write_page` method and
+/// information about the page size
+pub trait PageWrite<E> {
+    fn page_write(&mut self, address: u32, data: &[u8]) -> Result<(), Error<E>>;
+    fn page_size(&self) -> usize;
+}
```

Comment:
Since we will be making a breaking release due to the increase of the MSRV, we can increase it further to 1.51.0 and then we can use const generics.
Would you have interest in having a try at implementing this by converting the page size `Eeprom24x` parameter into a const parameter and using that instead of this trait?
The same can be applied to some other traits in this crate but this would be a start.
[Here](https://github.com/japaric/heapless/blob/7546c64e791e7d8f67d12b0a9fc74031ec1b8cba/src/string.rs#L8) is an example.